src/corecontext.h

```diff
@@ -0,0 +1,28 @@
+// As shown here: https://github.com/mortennobel/QtOpenGL3.2Core
+
+#ifndef DISPLAZ_CORECONTEXT_H
+#define DISPLAZ_CORECONTEXT_H
+
+#include <QGLContext>
+#include <QGLFormat>
+
+#if defined(Q_OS_MAC)
+#if QT_VERSION < 0x050000 && QT_VERSION >= 0x040800 // if less than Qt 5.0.0
+void* select_3_2_mac_visual(GDHandle handle, int depthBufferSize, bool multiSample=false);
+#endif
+#endif
+
+struct CoreContext : public QGLContext {
+    CoreContext(const QGLFormat& format, QPaintDevice* device) : QGLContext(format,device) {}
+    CoreContext(const QGLFormat& format) : QGLContext(format) {}
+
+#if defined(Q_OS_MAC)
+#if QT_VERSION < 0x050000 && QT_VERSION >= 0x040800 // if less than Qt 5.0.0
+    virtual void* chooseMacVisual(GDHandle handle) {
+        return select_3_2_mac_visual(handle, this->format().depthBufferSize()); //TODO: check if we want to enable multiSampling
+    }
+#endif
+#endif
+};
+
+#endif //DISPLAZ_CORECONTEXT_H
```

Comment:
I want displaz to work with Qt4 on linux if possible.  The question is whether I should bother maintaining the Qt4 compatibility for OS X as well.  For windows, I'm inclined just to drop Qt4 completely since it's actually easier for developers to build against Qt5 on that platform (in particular, there's no official Qt4 binaries for windows).

Is that also the situation on OS X?  If so, it seems better just to say "it doesn't work with Qt4 on OS X - build with Qt5 instead - it's easier anyway".